src/uwtools/tests/drivers/test_driver.py

```diff
@@ -83,13 +101,160 @@ def schema():
 """.strip()
 
 
+def test_create_directory_structure(tmp_path):
+    run_dir = tmp_path / "run"
+    with patch.object(Driver, "_create_run_directory") as _create_run_directory:
+        Driver.create_directory_structure(run_dir)
+    _create_run_directory.assert_called_once_with(run_dir, "delete")
+
+
+def test_create_directory_structure_bad_existing_act():
+    with raises(ValueError):
+        Driver.create_directory_structure(run_directory="/some/path", exist_act="foo")
+
+
+def test_run_cmd_expected(mpi_config, tmp_path):
+    config_file = tmp_path / "config.yaml"
+    YAMLConfig.dump_dict(config_file, mpi_config)
+    with patch.object(Driver, "_validate", return_value=True):
+        driver = ConcreteDriver(config_file=config_file)
+    driver._config = driver._experiment_config["component"]
+    assert driver.run_cmd() == "srun bar foo"
+
+
+def test_run_cmd_no_runtime_info(mpi_config, tmp_path):
+    config_file = tmp_path / "config.yaml"
+    del mpi_config["component"]["runtime_info"]
+    YAMLConfig.dump_dict(config_file, mpi_config)
+    with patch.object(Driver, "_validate", return_value=True):
+        driver = ConcreteDriver(config_file=config_file)
+    driver._config = driver._experiment_config["component"]
+    with raises(KeyError):
+        driver.run_cmd()
+
+
+def test_scheduler():
+    config_file = fixture_path("fruit_config.yaml")
+    with patch.object(Driver, "_validate", return_value=True):
+        concretedriver = ConcreteDriver(config_file=config_file)
+        # pylint: disable=pointless-statement
+        with patch.object(driver.JobScheduler, "get_scheduler") as get_scheduler:
+            concretedriver.scheduler
```

Comment:
If I'm reading right, `Driver.scheduler` is a `@property` method that returns the value returned by calling `JobScheduler.get_scheduler()`. We've mocked the latter, so we can control what it returns. It might be useful to demonstrate that `.scheduler` is essentially a wrapper around `.get_scheduler()`, which could maybe be done like
```
def test_scheduler():
    with patch.object(Driver, "_validate", return_value=True):
        concretedriver = ConcreteDriver(config_file=fixture_path("fruit_config.yaml"))
        with patch.object(concretedriver, "resources", return_value="bar"):
            with patch.object(driver.JobScheduler, "get_scheduler") as get_scheduler:
                get_scheduler.return_value = "foo"
                assert concretedriver.scheduler == "foo"
                get_scheduler.assert_called_once_with("bar")
```
That demonstrates that the value of `.scheduler` is based on the value returned by `.get_scheduler()`, and that `.get_scheduler()` gets called with whatever `.resources()` returns, which we can also control. No more `pointless-statement`.

Reply:
Or maybe we don't need to mock `.resources()` -- looks like we know it's going to return `{}` in our tests. So
```
ef test_scheduler():
    with patch.object(Driver, "_validate", return_value=True):
        concretedriver = ConcreteDriver(config_file=fixture_path("fruit_config.yaml"))
        with patch.object(driver.JobScheduler, "get_scheduler") as get_scheduler:
            get_scheduler.return_value = "foo"
            assert concretedriver.scheduler == "foo"
            get_scheduler.assert_called_once_with({})
```